src/main/webapp/video.js

```diff
@@ -0,0 +1,59 @@
+// TODO: Fetch video-id from servlet rather than URL.
+
+/* Used to gather URL parameters. */
+const URL_PARAM_ID = 'id';
+const URL_PARAM_VIDEO_ID = 'video-id';
+
+/* Database ID for specific lecture. */
+const LECTURE_ID = getLectureKey();
+/* Video ID for specific lecture. */
+const VIDEO_ID = getVideoId();
+
+/* TODO: Here for testing, remove. */
+console.log(LECTURE_ID);
+console.log(VIDEO_ID);
+
+/* Used to control video. */
+let videoPlayer;
+
+loadApi();
+
+/** Loads YouTube iFrame API. */
+async function loadApi() {
+  const videoApiScript = document.createElement('script');
+  const firstScriptTag = document.getElementsByTagName('script')[0];
+  videoApiScript.src = 'https://www.youtube.com/iframe_api';
+  firstScriptTag.parentNode.insertBefore(videoApiScript, firstScriptTag);
+}
+
+/**
+ * Creates a YouTube Video iFrame playing video with id:{@code VIDEO_ID} after
+ * API is loaded.
+ */
+function onYouTubeIframeAPIReady() {
+  videoPlayer = new window.YT.Player('player', {
+    height: '390',
+    width: '640',
+    videoId: VIDEO_ID,
+    events: {
+      onReady: onPlayerReady,
+    },
+  });
+}
+
+/** Plays video. Called after API and iFrame load. */
+function onPlayerReady(event) {
+  event.target.playVideo();
+}
+
+/** Returns database ID for lecture. */
+function getLectureKey() {
```

Comment:
This should stay in view.js, maybe also create a loadVideo() in here that view.js calls in `initialize`.  Sorry for being unclear, take a look at master to see how discussion.js and view.js interact, maybe merge them?

Reply:
As in merge master into this branch